EnergySage/es-ds
# feat: Add EsFileUpload example

### 🔗 Linked issue

#113

### ❓ Type of change




### 📚 Description





- Added an EsFileUpload example
- `EsFileInput`
    - Added a `deleteFileName` prop to delete a file from file list when the value changes
    - When emitting `fileDataRead`, create a URL for application type files
    - Changed `verifyFiles` to no longer overwrite the existing list of files but rather add to it
    - `uploadUrls` is now specific to a file name
    - When `readyToUpload` is emitted, the list of files (with all their data) is emitted instead of just the number of files
- `EsFileThumbnail`
    - Truncated display name
    - Emit the file name as payload for emitted events
- `mime-type-finder`
    - Add `humanReadable` field to mime types to allow for better error handling

#### Desktop
##### Normal state
<img width="1139" alt="Screenshot 2023-07-17 at 9 09 03 AM" src="https://github.com/EnergySage/es-ds/assets/43626328/815ee1ad-13d5-47ff-95b8-8eb19d1894d7">

##### With files
<img width="1139" alt="Screenshot 2023-07-17 at 9 09 28 AM" src="https://github.com/EnergySage/es-ds/assets/43626328/52cb24f4-48fd-4f23-a4db-27bd79b2be4b">

##### Failed states
<img width="1139" alt="Screenshot 2023-07-17 at 9 11 19 AM" src="https://github.com/EnergySage/es-ds/assets/43626328/ef28d50a-cd7a-4dea-9bcb-8360d04acfc0">
<img width="1139" alt="Screenshot 2023-07-17 at 9 11 53 AM" src="https://github.com/EnergySage/es-ds/assets/43626328/46c656dc-de2a-4d8e-95a0-5e19ca7678a9">
<img width="1139" alt="Screenshot 2023-07-17 at 9 12 10 AM" src="https://github.com/EnergySage/es-ds/assets/43626328/11287984-acd3-4b94-b6d3-1f7bce41022e">


#### Mobiles
##### Normal state
<img width="411" alt="Screenshot 2023-07-17 at 9 14 27 AM" src="https://github.com/EnergySage/es-ds/assets/43626328/de801970-3d65-4b52-9301-aa663af1d7c8">

##### With files
<img width="411" alt="Screenshot 2023-07-17 at 9 14 47 AM" src="https://github.com/EnergySage/es-ds/assets/43626328/ac058ee0-88ed-48b1-827f-82eafebf63a5">

##### Failed states
<img width="411" alt="Screenshot 2023-07-17 at 9 15 11 AM" src="https://github.com/EnergySage/es-ds/assets/43626328/c513f0d8-7f96-424e-a43e-882eeda6d23f">
<img width="411" alt="Screenshot 2023-07-17 at 9 15 43 AM" src="https://github.com/EnergySage/es-ds/assets/43626328/e7f1ee9e-344f-4130-95af-31c732712d2e">
<img width="411" alt="Screenshot 2023-07-17 at 9 16 00 AM" src="https://github.com/EnergySage/es-ds/assets/43626328/4cb9d321-b54b-4b30-8910-fd8cd95813da">

### 🥼 Testing



- Manual testing of components on mobile and desktop
- Tried to do all of the following:
    - Cause server error
    - Choose file that is too large (drag and drop)
    - Choose incorrect file type (drag and drop)
    - Upload multiple files
    - Upload single files
    - Delete files
    - Preview files
    - Upload, delete, then upload files
- For the demo, files are all uploaded to [energysage.free.beeceptor.com](https://energysage.free.beeceptor.com). You can see the console here: [beeceptor.com/console/energysage](https://beeceptor.com/console/energysage)

### 📝 Checklist





- [x] I have documented testing approach
